rocAL/rocAL/source/coco_meta_data_reader_key_points.cpp

```diff
@@ -0,0 +1,388 @@
+                _map_img_sizes.insert(pair<std::string, std::vector<ImgSize>>(image_name, img_sizes));
+                img_sizes.clear();
+            }
+        }
+        else if (0 == std::strcmp(key, "categories"))
+        {
+            RAPIDJSON_ASSERT(parser.PeekType() == kArrayType);
+            parser.EnterArray();
+            int id = 1, continuous_idx = 1;
+            while (parser.NextArrayValue())
+            {
+                if (parser.PeekType() != kObjectType)
+                {
+                    continue;
+                }
+                parser.EnterObject();
+                while (const char *internal_key = parser.NextObjectKey())
+                {
+                    if (0 == std::strcmp(internal_key, "id"))
+                    {
+                        id = parser.GetInt();
+                    }
+                    else
+                    {
+                        parser.SkipValue();
+                    }
+                }
+                _label_info.insert(std::make_pair(id, continuous_idx));
+                continuous_idx++;
+            }
+        }
+        else if (0 == std::strcmp(key, "annotations"))
+        {
+            RAPIDJSON_ASSERT(parser.PeekType() == kArrayType);
+            parser.EnterArray();
+            while (parser.NextArrayValue())
+            {
+                int id = 1, label = 0, is_crowd = 0;
+                float joint_sum = 0.0, area = 0.0;
+                long int ann_id = 0;
+                std::array<float, NUMBER_OF_JOINTS * 3> keypoint{};
+                if (parser.PeekType() != kObjectType)
+                {
+                    continue;
+                }
+                parser.EnterObject();
+                while (const char *internal_key = parser.NextObjectKey())
+                {
+                    if (0 == std::strcmp(internal_key, "image_id"))
+                    {
+                        id = parser.GetInt();
+                    }
+                    else if (0 == std::strcmp(internal_key, "category_id"))
+                    {
+                        label = parser.GetInt();
+                    }
+                    else if (0 == std::strcmp(internal_key, "id"))
+                    {
+                        ann_id = parser.GetDouble();
+                    }
+                    else if (0 == std::strcmp(internal_key, "is_crowd"))
+                    {
+                        is_crowd = parser.GetInt();
+                    }
+                    else if (0 == std::strcmp(internal_key, "area"))
+                    {
+                        area = parser.GetDouble();
+                    }
+                    else if (0 == std::strcmp(internal_key, "bbox"))
+                    {
+                        RAPIDJSON_ASSERT(parser.PeekType() == kArrayType);
+                        parser.EnterArray();
+
+                        box_center.push_back(parser.NextArrayValue() * parser.GetDouble());
```

Comment:
it is better to use arrays for data structures with fixed size like box_center, box_scale etc

Reply:
Done